pallets/nft/src/mock.rs

```diff
@@ -121,10 +122,37 @@ pub fn run_to_block(n: u32) {
 	}
 }
 
-pub fn events() -> Vec<Event> {
-	let evt = System::events().into_iter().map(|evt| evt.event).collect::<Vec<_>>();
+pub fn events_filter<T: 'static>() -> Vec<Event> {
+	let mut evt = System::events();
 
-	System::reset_events();
+	evt.retain(|evt| if_right_events::<T>(&evt.event));
+	return evt.into_iter().map(|evt| evt.event).collect::<Vec<_>>();
+}
+
+pub fn if_right_events<T: 'static>(evt: &Event) -> bool {
+	if TypeId::of::<T>() == TypeId::of::<Event>() {
+		return true;
```

Comment:
Under what case this is false and the next branch is evaluated true?

Reply:
Under two circumstance this function need to be updated.
(1) There is a significant change about Event class(Not crate's Event, but the Event create in the mock scope with Event::frame_system(raw_event), Event:pallet_balance(raw_event) usage. I have not figured out the true declare yet. But I guess it created under contruct_runtime!

(2) We add/remove test pallets, then  the match logic of if_right_events should change accordingly. I can not find a proper method to put enum's inside class as a function input.

Reply:
I will include this in file comment. But there might be "mistakes" since some mechs are not fully understood.